### PR TITLE
Migrate all Objective-C TUs to ARC.

### PR DESCRIPTION
**Pull request by chinmaygarde**
The RAII containers have been kept to make the migration easier. These container
are entirely redundant. That they are redundant is now being test by new
unit-tests in FML. The new unit-tests along with a few runs of the application
in Instruments yielded no notable difference in memory usage.

Fixes https://github.com/flutter/flutter/issues/38777

### CODE REVIEW

**Comment by chinmaygarde** [comment]
Needs buildroot patch https://github.com/flutter/buildroot/pull/304.

**Comment by stuartmorgan** [comment]
This is extremely hard to review as a giant PR, and it will also make pinpointing regressions unnecessarily difficult; I cannot recommend strongly enough against this approach.

Properly reviewing an ARC migration PR requires evaluating the *entire* file being changed, not just the lines touched. Validating that the files compile is trivial, but ensuring that, e.g., block capture semantics have been properly updated is not—a local that used to be a raw pointer is, when compiled under ARC, suddenly a strong pointer, which can have substantial implications. I've seen projects that have done large "mechanical" ARC changes introduce *massive* memory leaks that weren't caught until much later.

I think the approach of keeping scoped_* does not actually make this transition easier, since it encourages treating this as the mechanical change that it isn't, rather than the potentially subtle rewrite of every class that it actually is.

I am not comfortable reviewing this PR in this form, and I say that as someone who has spent most of my professional career writing ObjC code, both with and without ARC.

File: shell/platform/darwin/ios/framework/Source/FlutterPluginAppLifeCycleDelegate.mm [hunk 1]
```diff
@@ -37,7 +37,7 @@ - (void)addObserverFor:(NSString*)name selector:(SEL)selector {
   dispatch_block_t unsubscribe = ^{
     [[NSNotificationCenter defaultCenter] removeObserver:blockSelf name:name object:nil];
```
Comment (jmagman) [inline shell/platform/darwin/ios/framework/Source/FlutterPluginAppLifeCycleDelegate.mm]: To @stuartmorgan's point, this looks like a new retain cycle.  This object strongly retains _notificationUnsubscribers, which strongly retains blocks that strongly retain this object.

- Naive fix is to weakly retain self in the block.
- Better fix would be to convert _notificationUnsubscribers to a NSMutableArray<NSString *> of notification NAMES to remove the observe to in dealloc.
- Even better would be to just call [[NSNotificationCenter defaultCenter] removeObserver:self] in -dealloc which handles all notifications (though not needed at all in iOS 9.0 or macOS 10.11 and later).

**Comment by cbracken** [comment]
I'm with @stuartmorgan. I'd recommend we do this piecewise in some systematic way rather than as a mega-change. @chinmaygarde can you break this up into smaller patches that are easier to evaluate for reviewers?

**Pull request closed without merging**